src/worktree.ts

```diff
@@ -44,14 +46,22 @@ export async function generateWorktree(
       action.silent
     )
     const checkout = new GitCheckout(action.branch)
+
     if (branchExists) {
       // There's existing data on the branch to check out
       checkout.commitish = `origin/${action.branch}`
     }
+
     if (!branchExists || action.singleCommit) {
       // Create a new history if we don't have the branch, or if we want to reset it
       checkout.orphan = true
     }
+
+    if (action.isCrossRepositoryDeployment) {
+      // Used when cross repo deploying to ensure that context is focused on the correct remote.
+      await execute(`git fetch`, action.workspace, action.silent)
```

Comment:
I don't understand this one yet.

Is there a reason why we'd have to do this here instead of using the fetch before?

I remember that I've been surprised about the options for `checkout` based on the status quo. I guess this fetch does something to the status quo, but I can't nail it down.

This would be the situation where the cross-repo deployment doesn't have the target branch, right?

Reply:
I'll do some testing here to nail down what's happening here on Sunday, I'm constructing a new workstation but this is definitely my first order of business 👍